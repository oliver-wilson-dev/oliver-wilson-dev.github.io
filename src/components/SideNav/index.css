@import "../../styles/shared.css";

.crossIcon {
    position: absolute;
    top: 0;
    right: 0;
    margin: 10px;
    margin: var(--headerPadding);
    fill: var(--primaryTextColor);
    opacity: 0;

    transition-property: opacity;
    transition-duration: .3s;
    transition-timing-function: ease;
}

.crossIconShow {
    opacity: 1;
}

.menuContainer {
    width: 0px;
    background-color: transparent;

    height: 100vh;
    z-index: 5;
    position: absolute;
    top: 0;
    right: 0;

    display: flex;
    justify-content: flex-end;
    width: 100vw;
}

@keyframes open {
    0% {
        transform: translate(100vw);
        background-color: transparent;
    }

    50% {
        transform: translate(0);
        background-color: transparent;
    }

    100% {
        transform: translate(0);
        background-color: var(--overlayBackground);
    }
}

@keyframes close {
    0% {
        transform: translate(0);
        background-color: var(--overlayBackground);
    }

    50% {
        transform: translate(0);
        background-color: transparent;
    }

    100% {
        transform: translate(100vw);
        background-color: transparent;
    }
}

.menuOpen {
    animation-name: open;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal;
}

.menuClose {
    animation-name: close;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal
}

.menuHideFirstRender {
    display: none;
}

.menuContent {
    width: 80%;
    height: 100%;
    padding: 50px 0px;
    background-color: var(--primaryBackground);
    box-shadow: 0px 0px 8px var(--primaryBorderColourLight);
    
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    opacity: 1;
}

.menuList {
    width: 100%;
    padding: 0px;
    margin: 20px 0px;
    list-style-type: none;
    display: flex;
    flex-direction: column;
    max-height: 80%;
    overflow-y: scroll;
    overflow-x: hidden;
    justify-content: flex-start;
}

.menuItem {
    text-align: center;
    font-weight: 700;
    font-size: 24px;
    font-family: 'Source Sans Pro', sans-serif;

    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    &:hover, &:focus {
        background-color: var(--primaryBackgroundInverse);

        & .link {
            color: var(--primaryTextColorInverse);
        }
    }
}

.menuItemMargin {
    margin: 20px;
}

.menuItemNoHover {
    &:hover {
        background-color: var(--primaryBackground);
    }
}

.link {
    color: var(--primaryTextColor);
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

@media (min-width: 786px) {
    .menuContent {
        max-width: 600px;
    }

    .menuItem {
        font-size: 30px;
    }

    .menuItemMargin {
        margin: 40px;
    }
}