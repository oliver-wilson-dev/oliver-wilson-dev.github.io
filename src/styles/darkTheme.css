@import "./index.css";

[data-theme="DARK_THEME"] {
    --primaryBackground: #333;
    --primaryBackgroundImage: linear-gradient(#333, #333);
    --primaryBorderColour: #f8f8f8;
    --primaryTextColor: #f8f8f8;
    --spinnerTrailColour: rgba(248,248,248, 0.2);
    --sectionFadeColourFrom: rgba(51, 51, 51, 0);
    --sectionFadeColourTo: rgba(51, 51, 51, 1);
    --unvisitedLink: var(--projectGreen)
}