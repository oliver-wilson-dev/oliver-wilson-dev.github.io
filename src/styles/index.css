:root {
    --primaryBackgroundLight: #f8f8f8;
    --primaryBackgroundDark: #333;
    --primaryBackground: var(--primaryBackgroundLight);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: #333;

    --primaryTextColorDark: #f8f8f8;
    --primaryTextColorLight: #333;

    --primaryTextColorInverse: var(--primaryTextColorLight);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --linkBlue: rgb(21, 64, 204);
    --unvisitedLink: var(--linkBlue);

    --profilePictureBorder: var(--primaryBorderColour);

    --headerHeight: 65px;
    --headerPadding: 15px;

    --overlayBackground: rgba(0,0,0,.288);

    --projectGreen: #06d6a0;
    --projectPink: #EF476F;
    --projectYellow: #FFD166;
    --projectBlue: #118AB2;
    --projectDarkBlue: #073B4C;
}