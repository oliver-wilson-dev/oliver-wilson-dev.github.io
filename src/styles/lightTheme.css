@import "./index.css";

[data-theme="LIGHT_THEME"] {
    --primaryBackground: var(--primaryBackgroundLight);
    --primaryBackgroundInverse: var(--primaryBackgroundDark);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: var(--primaryTextColorLight);
    --primaryTextColorInverse: var(--primaryTextColorDark);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --primaryBackgroundLight: #f8f8f8;

    --unvisitedLink: var(--projectGreen);
    --unvisitedLinkInverse: var(--projectGreen);
}