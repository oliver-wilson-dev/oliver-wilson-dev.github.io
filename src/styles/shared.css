.bodyFixed {
    position: fixed;
    left: 0;
    width: 100%;
    overflow-y: scroll;
}

.pageHeader {
    font-size: 5em;
    margin: 0;
    margin-top: 10px;
    background-clip: text;
    -webkit-background-clip: text;
    letter-spacing: 2px;
    text-align: center;
    width: 100%;
  }

  @media (min-width: 786px) {
    .landing--header {
    max-width: 800px;
    }
  }

  @media (min-width: 0px) and (max-width: 767px) {
    .pageHeader {
      font-size: 3em;
    }
  }