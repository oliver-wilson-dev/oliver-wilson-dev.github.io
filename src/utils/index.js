export { preventScroll, allowScroll } from './overlayScroll';
