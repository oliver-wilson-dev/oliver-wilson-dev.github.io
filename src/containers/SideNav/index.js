import { connect } from 'react-redux';
import SideNav from '../../components/SideNav';
import { getShowSideNav } from '../../state/selectors';
import toggleShowSideNav from '../../state/actions/toggleShowSideNav';

const mapStateToProps = state => ({
  showSideNav: getShowSideNav(state)
});

const mapDispatchToProps = dispatch => ({
  toggleShowSideNav: () => dispatch(toggleShowSideNav())
});

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);
