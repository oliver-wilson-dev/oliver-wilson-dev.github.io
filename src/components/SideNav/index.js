/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import { Link } from 'react-router-dom';
import styles from './index.css';
import headerStyles from '../Header/index.css';

import CrossIcon from '../Icons/cross.svg';
import DownloadButton from '../DownloadButton';
import { preventScroll, allowScroll } from '../../utils';

const ignoreClick = (event) => {
  event.stopPropagation();
};

const SideNav = ({ showSideNav, toggleShowSideNav }) => {
  const isFirstRender = useRef(true);


  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else if (showSideNav) {
      preventScroll();
    } else {
      allowScroll();
    }
  }, [showSideNav]);

  const onOverlayClick = () => {
    toggleShowSideNav();
  };


  return (
    <div
      className={cn(styles.menuContainer, {
        [styles.menuOpen]: showSideNav,
        [styles.menuClose]: !showSideNav,
        [styles.menuHideFirstRender]: isFirstRender.current
      })}
      onClick={onOverlayClick}
    >
      <div
        onClick={ignoreClick}
        className={styles.menuContent}
      >
        <CrossIcon
          className={cn(styles.crossIcon, headerStyles.icon, {
            [styles.crossIconShow]: showSideNav
          })}
          onClick={toggleShowSideNav}
        />
        <ul className={styles.menuList}>
          <li className={styles.menuItem}>
            <Link to="/" className={cn(styles.link, styles.menuItemMargin)} onClick={toggleShowSideNav}>
              Home
            </Link>
          </li>
          <li className={styles.menuItem}>
            <Link to="/about" className={cn(styles.link, styles.menuItemMargin)} onClick={toggleShowSideNav}>
              About
            </Link>
          </li>
          <li className={cn(styles.menuItem, styles.menuItemMargin, styles.menuItemNoHover)}>
            <DownloadButton />
          </li>
        </ul>
      </div>
    </div>
  );
};

SideNav.propTypes = {
  showSideNav: PropTypes.bool.isRequired,
  toggleShowSideNav: PropTypes.func.isRequired,
};

export default SideNav;
