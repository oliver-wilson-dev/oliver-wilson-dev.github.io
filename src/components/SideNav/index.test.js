import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import SideNav from './index';

import styles from './index.css';
import { preventScroll, allowScroll } from '../../utils';

jest.mock('react-router-dom', () => ({ Link: ({ children }) => <button type="button">{children}</button> }));

jest.mock('../../utils', () => ({
  preventScroll: jest.fn(),
  allowScroll: jest.fn()
}));

jest.mock('../Icons/cross.svg', () => {
  const CrossIcon = () => null;

  return CrossIcon;
});

jest.mock('../DownloadButton', () => {
  const DownloadButton = () => null;

  return DownloadButton;
});

jest.mock('../../containers/ThemeToggle', () => {
  const ThemeToggle = () => null;

  return ThemeToggle;
});

jest.mock('../../containers/SideNav', () => {
  const SideNav = () => null;

  return SideNav;
});

const defaultProps = { showSideNav: false, toggleShowSideNav: jest.fn() };

const render = (
  overrideProps,
  renderMethod = shallow
) => renderMethod(<SideNav {...defaultProps} {...overrideProps} />);

describe('<SideNav/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when the showSideNav prop is true', () => {
    it('should open the sub navigation', () => {
      const component = render({ showSideNav: true });

      expect(component).toMatchSnapshot();
    });

    it('should call preventScroll', () => {
      const component = render({ showSideNav: false }, mount);

      component.setProps({ ...defaultProps, showSideNav: true });

      expect(preventScroll).toHaveBeenCalledTimes(1);
    });

    describe('and then clicking the overlay', () => {
      it('should call toggleShowSideNav to close the menu', () => {
        const component = render({ showSideNav: false }, mount);

        component.setProps({ ...defaultProps, showSideNav: true });

        act(() => {
          component.find(`.${styles.menuContainer}`).props().onClick();
        });

        expect(defaultProps.toggleShowSideNav).toHaveBeenCalledTimes(1);
      });
    });

    describe('and then clicking on the whitespace on the menu', () => {
      it('should not close the menu', () => {
        const mockEvent = {
          stopPropagation: jest.fn()
        };
        const component = render({ showSideNav: false }, mount);

        component.setProps({ ...defaultProps, showSideNav: true });

        act(() => {
          component.find(`.${styles.menuContent}`).simulate('click', mockEvent);
        });

        expect(component).toMatchSnapshot();
        expect(mockEvent.stopPropagation).toHaveBeenCalledTimes(1);
      });
    });

    describe('and then clicking a menu item', () => {
      it('should then close the menu', () => {
        const component = render({ showSideNav: false }, mount);

        component.setProps({ ...defaultProps, showSideNav: true });

        act(() => {
          component.find(`.${styles.link}`).first().props().onClick();
        });

        expect(component).toMatchSnapshot();
      });
    });

    describe('and then clicking the close icon', () => {
      it('should call allowScroll', () => {
        const component = render({ showSideNav: false }, mount);

        component.setProps({ ...defaultProps, showSideNav: true });

        component.update();

        act(() => {
          component.find(`.${styles.crossIcon}`).props().onClick();
        });

        expect(defaultProps.toggleShowSideNav).toHaveBeenCalledTimes(1);

        component.setProps({ ...defaultProps, showSideNav: false });

        component.update();

        expect(allowScroll).toHaveBeenCalledTimes(1);
      });
    });
  });
});
