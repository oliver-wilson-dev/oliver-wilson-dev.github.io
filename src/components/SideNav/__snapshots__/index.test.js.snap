// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav/> component should render correctly 1`] = `
<div
  className="menuContainer menuClose menuHideFirstRender"
  onClick={[Function]}
>
  <div
    className="menuContent"
    onClick={[Function]}
  >
    <CrossIcon
      className="crossIcon icon"
      onClick={[MockFunction]}
    />
    <ul
      className="menuList"
    >
      <li
        className="menuItem"
      >
        <Link
          className="link menuItemMargin"
          onClick={[MockFunction]}
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        className="menuItem"
      >
        <Link
          className="link menuItemMargin"
          onClick={[MockFunction]}
          to="/about"
        >
          About
        </Link>
      </li>
      <li
        className="menuItem menuItemMargin menuItemNoHover"
      >
        <DownloadButton />
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SideNav/> component when the showSideNav prop is true and then clicking a menu item should then close the menu 1`] = `
<SideNav
  showSideNav={true}
  toggleShowSideNav={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="menuContainer menuOpen"
    onClick={[Function]}
  >
    <div
      className="menuContent"
      onClick={[Function]}
    >
      <CrossIcon
        className="crossIcon icon crossIconShow"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <ul
        className="menuList"
      >
        <li
          className="menuItem"
        >
          <Link
            className="link menuItemMargin"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            to="/"
          >
            <button
              type="button"
            >
              Home
            </button>
          </Link>
        </li>
        <li
          className="menuItem"
        >
          <Link
            className="link menuItemMargin"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            to="/about"
          >
            <button
              type="button"
            >
              About
            </button>
          </Link>
        </li>
        <li
          className="menuItem menuItemMargin menuItemNoHover"
        >
          <DownloadButton />
        </li>
      </ul>
    </div>
  </div>
</SideNav>
`;

exports[`<SideNav/> component when the showSideNav prop is true and then clicking on the whitespace on the menu should not close the menu 1`] = `
<SideNav
  showSideNav={true}
  toggleShowSideNav={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="menuContainer menuOpen"
    onClick={[Function]}
  >
    <div
      className="menuContent"
      onClick={[Function]}
    >
      <CrossIcon
        className="crossIcon icon crossIconShow"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <ul
        className="menuList"
      >
        <li
          className="menuItem"
        >
          <Link
            className="link menuItemMargin"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            to="/"
          >
            <button
              type="button"
            >
              Home
            </button>
          </Link>
        </li>
        <li
          className="menuItem"
        >
          <Link
            className="link menuItemMargin"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            to="/about"
          >
            <button
              type="button"
            >
              About
            </button>
          </Link>
        </li>
        <li
          className="menuItem menuItemMargin menuItemNoHover"
        >
          <DownloadButton />
        </li>
      </ul>
    </div>
  </div>
</SideNav>
`;

exports[`<SideNav/> component when the showSideNav prop is true should open the sub navigation 1`] = `
<div
  className="menuContainer menuOpen menuHideFirstRender"
  onClick={[Function]}
>
  <div
    className="menuContent"
    onClick={[Function]}
  >
    <CrossIcon
      className="crossIcon icon crossIconShow"
      onClick={[MockFunction]}
    />
    <ul
      className="menuList"
    >
      <li
        className="menuItem"
      >
        <Link
          className="link menuItemMargin"
          onClick={[MockFunction]}
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        className="menuItem"
      >
        <Link
          className="link menuItemMargin"
          onClick={[MockFunction]}
          to="/about"
        >
          About
        </Link>
      </li>
      <li
        className="menuItem menuItemMargin menuItemNoHover"
      >
        <DownloadButton />
      </li>
    </ul>
  </div>
</div>
`;
